docs/source/guides/integrations.mdx

```diff
@@ -0,0 +1,170 @@
+# Integrate any ML framework with the Hub
+
+The Huggingface Hub makes it easy to host and share models with the community. It already has support for
+[dozens of libraries](https://huggingface.co/docs/hub/models-libraries) in the Open Source ecosystem. We are always
+working on expanding this support to push Machine Learning forward. The `huggingface_hub` library plays a key role in
+this process, allowing any Python script to easily push and load files.
+
+When talking about integrating a library to the Hub, we have 4 main topics:
+1. **Push to Hub:** implement a `push_to_hub()`-like method to upload a model to the Hub.
+   This includes the model weights as well as the modelcard and any other relevant information/data necessary to run the model (example: logs from training).
+2. **Download from Hub:** implement a `from_pretrained`-like method to load a model from the Hub.
+   Takes care of downloading the model configuration/weights and instantiating it.
+3. **Inference API:** collaborate with the HF Team to make it possible to run inference on our servers, using your library.
+4. **Widgets:** collaborate with the HF Team to display a widget on the landing page of your models on the Hub. It allows users to quickly try a model from the browser.
+
+In this guide, we will focus on topic 1 and 2. If you are interested in Inference and Widgets, please let us know!
+You can also reach out to us if you are integrating your library with the Hub and want to be listed [in our docs](https://huggingface.co/docs/hub/models-libraries).
+
+## A bit of theory: use class inheritance
+
+As we saw above, we are interested in having 2 methods: one to upload files (`push_to_hub`) and one to download files (`from_pretrained`).
+In most cases, a library already implements its model using a Python class. The class contains the properties of the model
+and methods to load, run, train, evaluate,... it. It would be nice to extend the class with the 2 new methods!
+
+The recommended approach to do so is to use inheritance, and more precisely mixins. A [Mixin](https://stackoverflow.com/a/547714)
```

Comment:
Do we want to make this the official recommended approach? This is more complex/scary than the usually push to hub/download_from_hub with the HTTP endpoints and requires having classes in their libraries